diff --git a/substances/src/main/java/gsrs/ncats/substances/SubstancesApplication.java b/substances/src/main/java/gsrs/ncats/substances/SubstancesApplication.java
index e1c8f50..72185ec 100644
--- a/substances/src/main/java/gsrs/ncats/substances/SubstancesApplication.java
+++ b/substances/src/main/java/gsrs/ncats/substances/SubstancesApplication.java
@@ -16,6 +16,7 @@ import gsrs.EnableGsrsLegacyPayload;
 import gsrs.EnableGsrsLegacySequenceSearch;
 import gsrs.EnableGsrsLegacyStructureSearch;
 import gsrs.EnableGsrsScheduler;
+import gsrs.EnableGsrsJoseCryptoService;
 import gsrs.cv.EnableControlledVocabulary;
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
@@ -31,6 +32,7 @@ import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 @EntityScan(basePackages ={"ix","gsrs", "gov.nih.ncats"} )
 //@EnableJpaRepositories(basePackages ={"ix","gsrs", "gov.nih.ncats"} )
 @EnableGsrsScheduler
+@EnableGsrsJoseCryptoService
 @EnableGsrsBackup
 @EnableControlledVocabulary
 @EnableAsync
