diff --git a/gateway/src/main/java/gsrs/ncats/gateway/ZuulConfiguration.java b/gateway/src/main/java/gsrs/ncats/gateway/ZuulConfiguration.java
index 96736ce..76ccb75 100644
--- a/gateway/src/main/java/gsrs/ncats/gateway/ZuulConfiguration.java
+++ b/gateway/src/main/java/gsrs/ncats/gateway/ZuulConfiguration.java
@@ -2,6 +2,7 @@ package gsrs.ncats.gateway;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
+import java.util.TreeMap;
 
 import org.springframework.beans.BeansException;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -13,6 +14,7 @@ import org.springframework.cglib.proxy.Enhancer;
 import org.springframework.cglib.proxy.MethodInterceptor;
 import org.springframework.cglib.proxy.MethodProxy;
 import org.springframework.cglib.proxy.NoOp;
+import org.springframework.cloud.netflix.zuul.filters.ZuulProperties;
 import org.springframework.cloud.netflix.zuul.filters.RouteLocator;
 import org.springframework.cloud.netflix.zuul.web.ZuulController;
 import org.springframework.cloud.netflix.zuul.web.ZuulHandlerMapping;
@@ -26,6 +28,14 @@ public class ZuulConfiguration {
     /** The path returned by ErrorContoller.getErrorPath() with Spring Boot < 2.5 (and no longer available on Spring Boot >= 2.5). */
     private static final String ERROR_PATH = "/error";
 
+    private ZuulProperties zuulProperties;
+
+    @Autowired
+    public void setZuulProperties(ZuulProperties zuulProperties) {
+        zuulProperties.setRoutes(new TreeMap(zuulProperties.getRoutes()));
+        this.zuulProperties = zuulProperties;
+    }
+
     /**
      * Constructs a new bean post-processor for Zuul.
      * 
